v2/contribute/decisions/oauth/0010-m2m-admin-access.mdx

```diff
@@ -0,0 +1,46 @@
+---
+id: "0010"
+title: Recommend access tokens using custom scopes for M2M
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Recommend access tokens using custom scopes for M2M
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+We want to remove/simplify the M2M guide that uses the JWT recipe.
+
+## Considered Options
+
+* Provide separate API for admin keys
+* Recommend access tokens using custom scopes for M2M
+
+## Decision Outcome
+
+Chosen option: Recommend access tokens using custom scopes for M2M
+
```

Comment:
is this m2m between client backend and the oauth server? Or client backend to another client backend? Is there any distinction that needs to be made between these two?

Reply:
I ask cause I assume the option of "Provide separate API for admin keys" doesn't really apply to m2m between client backend to another client backend right?

Reply:
1. I don't think we need to make a distinction. The SDK is not exposing special endpoints on the oauth server that is accessible by oauth access tokens
2. Separate API for admin keys can apply to both. It'd have to be a separate recipe for the express purpose of authenticating M2M requests (both on the requester and the requested sides)

Reply:
> The SDK is not exposing special endpoints on the oauth server that is accessible by oauth access tokens

Shouldn't it? For example to get the profile info of the user with the access token? So if a flow doesn't request openid scope, then how will the client's app get the user's information?

> Separate API for admin keys can apply to both. It'd have to be a separate recipe for the express purpose of authenticating M2M requests (both on the requester and the requested sides)

I don't want to add any new recipe for this - it should all be do-able with oauth 2 recipe ideally

Reply:
> Shouldn't it?

We can (and should) expose a profile endpoint that returns whatever we'd add to the id token for an access token. I meant using the M2M access tokens. The profile endpoint does not really make sense using a token without an associated user (created by the client credentials flow).

Reply:
Right yea. This conversation diverged into another topic, but we should add a profile endpoint in the backend SDK then which does OAuth 2.0 access token validation right? Should this endpoint also be able to be queried by the session recipe's access token as well?

Reply:
> we should add a profile endpoint in the backend SDK then which does OAuth 2.0 access token validation right

Yes, I'll add this to the BE SDK interface.

> Should this endpoint also be able to be queried by the session recipe's access token as well?

We can create a similar one but I don't think we should mix the two together as it will create confusion (e.g.: how do scopes relate to session recipe tokens?)

Reply:
Agreed.